proxy/src/main/java/com/wavefront/agent/listeners/ZipkinPortUnificationHandler.java

```diff
@@ -0,0 +1,260 @@
+      BytesDecoder<zipkin2.Span> decoder = SpanBytesDecoderDetector.decoderForListMessage(bytesArray);
+      List<zipkin2.Span> zipkinSpanSink = new ArrayList<>();
+      decoder.decodeList(bytesArray, zipkinSpanSink);
+      processZipkinSpans(zipkinSpanSink);
+      status = HttpResponseStatus.ACCEPTED;
+      processedBatches.inc();
+    } catch (Exception e) {
+      failedBatches.inc();
+      writeExceptionText(e, output);
+      status = HttpResponseStatus.BAD_REQUEST;
+      logger.log(Level.WARNING, "Zipkin Thrift batch processing failed", Throwables.getRootCause(e));
+    }
+    writeHttpResponse(ctx, status, output, incomingRequest);
+  }
+
+  private void processZipkinSpans(List<zipkin2.Span> zipkinSpans) {
+    if (traceDisabled.get()) {
+      if (warningLoggerRateLimiter.tryAcquire()) {
+        logger.info("Ingested spans discarded because tracing feature is not enabled on the server");
+      }
+      discardedBatches.inc();
+      discardedTraces.inc(zipkinSpans.size());
+      return;
+    }
+
+    for (zipkin2.Span zipkinSpan : zipkinSpans) {
+      processZipkinSpan(zipkinSpan);
+    }
+  }
+
+  private void processZipkinSpan(zipkin2.Span zipkinSpan) {
+    List<Annotation> annotations = addAnnotations(zipkinSpan);
+
+    // TODO: Confirm source value
+    String sourceName = zipkinSpan.tags().get(SOURCE_KEY) == null ? "unknown" : zipkinSpan.tags().get(SOURCE_KEY);
+    annotations.add(new Annotation(SOURCE_KEY, sourceName));
+
+    //Build wavefront span
+    Span newSpan = Span.newBuilder()
+            .setCustomer("dummy")
+            .setName(zipkinSpan.name())
+            .setSource(sourceName)
+            .setSpanId(getSpanUuid(zipkinSpan))
+            .setTraceId(convertToUuidString(zipkinSpan.traceId()))
+            .setStartMillis(zipkinSpan.timestampAsLong() / 1000)
+            .setDuration(zipkinSpan.durationAsLong() / 1000)
+            .setAnnotations(annotations)
+            .build();
+    handler.report(newSpan);
+  }
+
+    private List<Annotation> addAnnotations(zipkin2.Span zipkinSpan) {
+    List<Annotation> annotations = new ArrayList<>();
+
+    // Set Span's Application Tags.
+    String applicationName = zipkinSpan.tags().get(APPLICATION_KEY) == null ? DEFAULT_APPLICATION : zipkinSpan.tags().get(APPLICATION_KEY);
+    annotations.add(new Annotation(APPLICATION_KEY, applicationName));
+
+    annotations.add(new Annotation(SERVICE_KEY, zipkinSpan.localServiceName()));
+
+    if (zipkinSpan.tags().get(CLUSTER_KEY) != null) {
+      annotations.add(new Annotation(CLUSTER_KEY, zipkinSpan.tags().get(CLUSTER_KEY)));
+    }
+    if (zipkinSpan.tags().get(SHARD_KEY) != null) {
+      annotations.add(new Annotation(SHARD_KEY, zipkinSpan.tags().get(SHARD_KEY)));
+    }
+
+    // Set Span's References.
+    if (zipkinSpan.parentId() != null) {
+      annotations.add(new Annotation("parent", convertToUuidString(zipkinSpan.parentId())));
```

Comment:
Can you put "parent" and "followsFrom" in a constants file so that Jaeger and Zipkin can use those constants?